platform/mime/core/platform-mime-core-impl/src/test/java/ddf/mime/mapper/MimeTypeMapperTest.java

```diff
@@ -123,4 +132,48 @@ public void testGuessMimeTypeForXmlIngest() throws Exception {
         assertNull(mimeType);
     }
 
+    @Test
+    public void testGuessMimeTypeForXmlIngestNoExtension() throws Exception {
+        MimeTypeMapper mapper = new MimeTypeMapperImpl(MOCK_MIME_TYPE_RESOLVERS);
+
+        InputStream is = FileUtils.openInputStream(new File(CSW_RECORD_FILE_NO_EXTENSION));
+        String mimeType = mapper.guessMimeType(is, "");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml;id=csw", mimeType);
+
+        is = FileUtils.openInputStream(new File(XML_METACARD_FILE_NO_EXTENSION));
+        mimeType = mapper.guessMimeType(is, "");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml", mimeType);
+    }
+
+    @Test
+    public void testGuessMimeTypeForFileExtension() throws Exception {
+        MimeTypeMapper mapper = new MimeTypeMapperImpl(MOCK_MIME_TYPE_RESOLVERS);
+
+        String mimeType = mapper.getMimeTypeForFileExtension("xml");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml;id=csw", mimeType);
+
+        mimeType = mapper.getMimeTypeForFileExtension(".nitf");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("image/nitf", mimeType);
+
+        //change the priority order to get a different resolver for .xml
+        List<MimeTypeResolver> resolversWithXmlPriority = new ArrayList<>();
+        resolversWithXmlPriority.add(new MockMimeTypeResolver("XmlMetacardResolver",
+                15,
+                new String[] {"xml=text/xml"},
+                "urn:catalog:metacard"));
+        resolversWithXmlPriority.add(new MockMimeTypeResolver("CswResolver",
+                10,
+                new String[] {"xml=text/xml;id=csw"},
+                "http://www.opengis.net/cat/csw/2.0.2"));
+
+        mapper = new MimeTypeMapperImpl(resolversWithXmlPriority);
+        mimeType = mapper.getMimeTypeForFileExtension("xml");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml", mimeType);
```

Comment:
This test adds two new resolvers with different priorities, starting on line 163. Line 160 might be a good place to end the test method and start a new one. 

Also, I can't really follow what lines 163 to the end are testing. Why does one resolver use "xml=text/xml" as the schema and the other resolver add ";id=csw" to the end of the schema? Using dummy values like
`new String[] {"xml=HIGH_PRIORITY"}`
and `new String[] {"xml=LOW_PRIORITY"}`
would be good.

Is the schema value (4th constructor parameter of the resolver) being used ? It  doesn't look like it. If it is not used, let's pass `null` or an empty string for the schema. Having real values made me think they were being used by the test somehow.